src/lib/containers/FilePreview/FileHandleContentRenderer.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { useGetPresignedUrlContent } from '../../utils/hooks/SynapseAPI/file/useFiles'
+import { MB } from '../../utils/SynapseConstants'
+import {
+  FileHandle,
+  FileHandleAssociation,
+  BatchFileRequest,
+} from '../../utils/synapseTypes'
+import { SynapseSpinner } from '../LoadingScreen'
+import HtmlPreview from './HtmlPreview/HtmlPreview'
+import { PreviewRendererType } from './PreviewRendererType'
+
+const MAX_FILE_SIZE = 10 * MB
+
+export type FileHandleContentRendererProps = {
+  /** The file handle whose contents should be downloaded and rendered */
+  fileHandle: FileHandle
+  /** The association between the file handle and an object which will give the user permission to access the file data */
+  fileHandleAssociation: FileHandleAssociation
+  /** Informs how to render the file data */
+  previewType: PreviewRendererType
+}
+
+/**
+ * Fetches the content for and renders the contents of a file handle.
+ * @param props
+ * @returns
+ */
+export default function FileHandleContentRenderer(
+  props: FileHandleContentRendererProps,
+) {
+  const { fileHandle, fileHandleAssociation, previewType } = props
+
+  const batchFileRequest: BatchFileRequest = {
+    requestedFiles: [fileHandleAssociation],
+    includePreSignedURLs: true,
+    includeFileHandles: false,
+    includePreviewPreSignedURLs: false,
+  }
+
+  const { data: content, isLoading } = useGetPresignedUrlContent(
+    fileHandle,
```

Comment:
While it would be nice to abstract the details of the request away, I don't want the component or the hook to be FileEntity only, so it could be reused in other places where we might need to download file handle contents (e.g. Wiki attachments).